input.cpp

```diff
@@ -5121,10 +5121,10 @@ int input_test(int getchar)
 								continue;
 							}
 
-							int xval, yval, zval;
-							xval = ((data[0] & 0x10) ? -256 : 0) | data[1];
-							yval = ((data[0] & 0x20) ? -256 : 0) | data[2];
-							zval = ((data[3] & 0x80) ? -256 : 0) | data[3];
+							// The bytes in the mouse delta data are signed
```

Comment:
This is not correct. offsets are 9 bits, not 8 bits. If mouse will give large offset then this code won't work. For example +128 is a legit value while your "fix" will mistreat it as -128.

Reply:
It's worth pointing out that this wasn't actually part of the "fix", it just seemed like a simplification (but if values are truly 9-bits, I agree it's wrong). The fix for my problem came from the other change of clamping deltas to -127 to 127 (but as stated, that could be buggy cores).